VultisigApp/VultisigApp/Services/Ton/TonService.swift

```diff
@@ -128,10 +129,54 @@ class TonService {
         if let balance = Utils.extractResultFromJson(fromData: data, path: "balance") as? String {
             return balance
         }
-                
+        
         return .zero
     }
-
+    
+    func getJettonBalance(_ coin: Coin) async throws -> String {
+        
+        guard let url = URL(string: Endpoint.fetchTonJettonBalance(address: coin.address, jettonAddress: coin.contractAddress)) else {
+            throw URLError(.badURL)
+        }
+        let request = URLRequest(url: url)
+        
+        let (data, _) = try await URLSession.shared.data(for: request)
+        
+        // Parse TonAPI response format
+        if let json = try? JSONSerialization.jsonObject(with: data) as? [String: Any],
```

Comment:
can you define a struct , and json decode it appropriately , don't use [String: Any] for that